tool/src/dart/mod.rs

```diff
@@ -248,23 +246,62 @@ impl<'a, 'cx> TyGenContext<'a, 'cx> {
                     annotation,
                     ffi_cast_type_name,
                     dart_type_name,
-                    get_expression,
-                    set_expressions,
+                    c_to_dart,
+                    dart_to_c,
                 }
             })
             .collect::<Vec<_>>();
 
-        let methods = ty
+        let mut methods = ty
             .methods
             .iter()
             .flat_map(|method| self.gen_method_info(id, method, type_name))
             .collect::<Vec<_>>();
 
+        // Non-out structs need to be constructible in Dart
+        let default_constructor = if !is_out {
+            if let Some(constructor) = methods
+                .iter_mut()
+                .find(|m| m.declaration.contains(&format!("{type_name}()")))
+            {
+                // If there's an existing zero-arg constructor, we repurpose it with optional arguments for all fields
```

Comment:
issue: I would like this behavior to be opt-in, perhpas `#[diplomat::attr(dart, factory)]`

but it's fine for now

Reply:
or really, like I said earlier, I want it to be such that people do not need to call FFI to construct these

Reply:
Once we have constructor things in Diplomat we can do that. Currently I'm heuristically detecting constructors for Dart